jdrfandroidbleparser/src/main/java/org/ehealthinnovation/jdrfandroidbleparser/bgm/characteristic/compoundcharacteristic/GlucoseMeasurementCharacteristic.kt

```diff
@@ -0,0 +1,137 @@
+package org.ehealthinnovation.jdrfandroidbleparser.bgm.characteristic.compoundcharacteristic
+
+import android.bluetooth.BluetoothGattCharacteristic
+import android.util.Log
+import org.ehealthinnovation.jdrfandroidbleparser.bgm.characteristic.BaseCharacteristic
+import org.ehealthinnovation.jdrfandroidbleparser.bgm.encodedvalue.bgmmeasurement.Flags
+import org.ehealthinnovation.jdrfandroidbleparser.bgm.encodedvalue.bgmmeasurement.SampleLocation
+import org.ehealthinnovation.jdrfandroidbleparser.bgm.encodedvalue.bgmmeasurement.SensorStatusAnnunciation
+import org.ehealthinnovation.jdrfandroidbleparser.bgm.encodedvalue.bgmmeasurement.Type
+import org.ehealthinnovation.jdrfandroidbleparser.bgm.utility.BluetoothDateTime
+import org.ehealthinnovation.jdrfandroidbleparser.encodedvalue.GattCharacteristic
+import org.ehealthinnovation.jdrfandroidbleparser.encodedvalue.Units
+import java.util.*
+
+class GlucoseMeasurementCharacteristic(characteristic: BluetoothGattCharacteristic?) : BaseCharacteristic(characteristic, GattCharacteristic.GLUCOSE_MEASUREMENT.assigned) {
+    var flags: EnumSet<Flags>? = null
+    var sequenceNumber: Int? = null
+    /**
+     * The year, month, day, hour, minute, second have value according to https://www.bluetooth.com/specifications/gatt/viewer?attributeXmlFile=org.bluetooth.characteristic.date_time.xml
+     * These are the raw value extracted from the binary packet and not directly usable without proper conversion.
+     * These values are converted into a [Date] as [displayTime]
+     */
+    var timeOffset: Int? = null
+    var timeYear: Int? = null
+    var timeMonth: Int? = null
+    var timeDay: Int? = null
+    var timeHour: Int? = null
+    var timeMinute: Int? = null
+    var timeSecond: Int? = null
+    var displayTime: Date? = null
```

Comment:
If you make the data class, as I suggested below, you can take these out of the characteristic, and just store the data class here. Will look cleaner.